km_api/account/serializers.py

```diff
@@ -197,9 +197,11 @@ def validate(self, data):
                 associated with the confirmation.
         """
         confirmation = models.EmailConfirmation.objects.get(key=data['key'])
-        user = authenticate(
-            email=confirmation.email.email,
-            password=data['password'])
+
+        email = confirmation.email.email
+        user = models.User.objects.get(email=email)
+
+        assert user.check_password(data['password']), user is None
```

Comment:
If the user's password doesn't match, an `AssertionError` will be raised. This is not desired because it won't be caught by the serializer and will cause the request to return a `500 Server Error` response.